src/components/ReceiptModal/ReceiptModal.jsx

```diff
@@ -49,24 +57,96 @@ export default function ReceiptModal(props) {
         setCurImageFile(img)
     }
 
+    const handleReceiptRemove = () => {
+        setCurImageUrl("")
+        setCurImageFile(null)
+        setLoadingPercent(0)
+    }
+
     const handleScanReceipt = async () => {
         try {
+            setScanningImage(true)
             let imageObject = {
                 url: curImageUrl
             }
-            axios.post(receiptScannerUrl, imageObject).then((res) => {
+            await axios.post(receiptScannerUrl, imageObject).then((res) => {
                 console.log(res.data)
+                const curItems = res.data.Items.values.map((val) => {
+                    let item = {
+                        name: val?.properties?.Description?.value || "",
+                        quantity: val?.properties?.Quantity?.value || "",
+                        unit: basicUnits[0],
+                        category: basicCategories[0]
+                    }
+                    return item
+                })
+                setReceiptItems(curItems)
             }).catch((error) => {
                 console.log(error.message)
             })
         } catch (error) {
             console.log(error.message)
         }
+        setScanningImage(false)
+    }
+
+    const handleReceiptOutputFormChange = (event, indx) => {
+        const curKey = event.target.name
+        const newItems = receiptItems.map(i => ({ ...i }))
+        newItems[indx][curKey] = event.target.value
+        setReceiptItems(newItems)
+    }
+
+    const handleEachItemUpdate = (item, products) => {
+        let itemIndex = products.findIndex(val => val.name === item.name)
+        if (itemIndex == -1) {
+            products.push(item)
+        } else {
+            if (products[itemIndex].unitType != item.unitType) {
+                setError("please make sure measurement is correct")
```

Comment:
When would this happen?

Reply:
it would be when the user puts in a weight unit (grams) when the product in the pantry has a volume unit (liters). I'm thinking instead of doing one of two options 1) if user puts in two different unit types, display both ("2 grams, 5 liters"), or 2) I could have an extra conversion that converts it to one (altho the conversion won't be accurate with changing density